getting-started/index.rst

```diff
@@ -78,19 +78,86 @@ using:
+from another platform or a release which uses the same Unicode version.  You
+can avoid needing most of the documentation tools by running configure with
+the ``--disable-documentation`` option.
+
+On other platforms
+~~~~~~~~~~~~~~~~~~
+
+You will need the following tools installed to build from git:
+
+* GNU m4 >= 4.6 (for autoconf)
+* perl >= 5.6 (for automake; also for various maintainer scripts)
+* python >= 2.3 (for generating the Python bindings)
+* GNU make (or another make which support VPATH for explicit rules)
+* GNU bison (for building SWIG, used for generating the bindings)
+* Tcl (to generate unicode/unicode-data.cc)
+
+There are also a number of libraries you'll need available, including
+development headers:
+
+* `zlib1g <https://zlib.net>`_
+* `libuuid <https://git.kernel.org/pub/scm/utils/util-linux/util-linux.git/tree/libuuid>`_
+* `PCRE <https://www.pcre.org>`_
+* libmagic (which is part of `the open source implementation of file(1) <https://www.darwinsys.com/file/>`_)
+
+.. On Fedora, yum install libuuid-devel -- can we get a more complete list?
+
+If you're doing much development work, you'll probably also want the following
+tools installed:
+
+* `valgrind <http://valgrind.org/>`_ for better testsuite error finding
+* `ccache <https://ccache.dev>`_ for faster rebuilds
+* `eatmydata <https://www.flamingspork.com/projects/libeatmydata/>`_ for faster testsuite runs
+
+If you want to be able to build distribution tarballs (with ``make dist``) then
+you'll also need some further tools:
+
+* doxygen (v1.8.8 is used for 1.3.x snapshots and releases; 1.7.6.1 fails to
+  process trunk after ``PL2Weight`` was added).
+* dot (part of Graphviz.  Doxygen's ``DOT_MULTI_TARGETS`` option apparently needs
+  ">1.8.10")
+* help2man
+* rst2html or rst2html.py (``pip install docutils``)
+* pngcrush (optional - used to reduce the size of PNG files in the HTML
+  apidocs)
+* sphinx-doc (``pip install sphinx`` should do)
 
 Building Xapian
 ---------------
 
 Bootstrapping the code
 ~~~~~~~~~~~~~~~~~~~~~~
 
-Xapian needs to set up a few things with a fresh clone of the code, as
-well as downloading and building some tools for which we require very
-precise versions. You should run this command in the ``xapian``
-directory that was created earlier when you cloned the source code:
+The repository does not contain any automatically generated files
+(such as ``configure``, ``Makefile.in``, Snowball-generated stemmers, Lemon-generated
+parsers, SWIG-generated code, and so on) because experience shows it's best to keep
+these out of version control.  To avoid requiring you to install the correct
+versions of the tools required, we either include the source to these tools in
+the repo directly (in the case of Snowball and Lemon), or the bootstrap script
+will download them as tarballs (autoconf, automake, libtool) or
+from git (SWIG), build them, and install them within the source tree.
```

Comment:
This is more helpful, but it would be good to clearly say somewhere that `bootstrap` (or more precisely the top-level `configure`) also hooks up the tree such that the in-tree build of xapian-core is used by the other directories.  This is something which GSoC applicants quite often seem to hit when they rerun `configure` in a subdirectory and end up using the libxapian-dev or equivalent package from their distro, and it's quite confusing if you don't know what's going on.

Reply:
👍